@here/harp-mapview/lib/geometry/TileGeometryCreator.ts

```diff
@@ -387,7 +391,7 @@ export class TileGeometryCreator {
         const materials: THREE.Material[] = [];
         const mapView = tile.mapView;
         const dataSource = tile.dataSource;
-        const zoomLevel = mapView.zoomLevel;
+        const displayZoomLevel = Math.floor(mapView.zoomLevel);
```

Comment:
```suggestion
        const zoomLevel = mapView.zoomLevel;
        const displayZoomLevel = Math.floor(zoomLevel);
```
There's no need to remove the ``zoomLevel`` variable here and the patch-set would be smaller that way, since you wouldn't need to update lines from ``zoomLevel`` to ``mapView.zoomLevel``, like...

Reply:
There are two cases in this method, where zoomLevel is used:
 1. creating objects
 1. pre-render update functions

in first case zoomLevel is used as integer, flooring current value down, passing clear (and persistent) level to properties that will not update further.
in second - we need up-to-date zoomLevel and can't use value from cached variable.

Reply:
You're, right didn't think about this case. Thanks for the explanation!